x86_symbolic/src/Data/Macaw/X86/Crucible.hs

```diff
@@ -131,14 +132,32 @@ withConcreteCountAndDir state val_size wrapped_count _wrapped_dir func =
 
 -- | An 'Exception' thrown when Crucible attempts to execute an instruction for
 -- which we have no semantics. The 'String' fields are human-readable messages.
-data MissingSemantics
-  = MissingStmtSemantics String
-  | MissingTermSemantics String
-  | MissingPrimFnSemantics String
-  deriving Show
+data MissingSemantics where
+  MissingPrimFnSemantics :: forall sym mt. M.X86PrimFn (AtomWrapper (RegEntry sym)) mt -> MissingSemantics
+  MissingStmtSemantics :: forall sym. M.X86Stmt (AtomWrapper (RegEntry sym)) -> MissingSemantics
+  MissingTermSemantics :: forall sym. M.X86TermStmt (AtomWrapper (RegEntry sym)) -> MissingSemantics
+
+instance Show MissingSemantics where
+  show = show . pretty
 
 instance Exception MissingSemantics
 
+instance Pretty MissingSemantics where
+  pretty =
+    \case
+      MissingPrimFnSemantics fn ->
+        pretty "Symbolic execution semantics for x86 primitive function are not implemented yet:"
+          <+> I.runIdentity (MC.ppArchFn (I.Identity . liftAtomIn (pretty . regType)) fn)
```

Comment:
This line is mostly copy-pasted from various spots, not sure it's correct but it does typecheck :shrug: